### PR TITLE
[Look&Feel] Revert tooltip addition for navigation menu

### LINKED ISSUE #7483: [BUG] The hamburger icon will always pop up tooltip when switching app
**Describe the bug**

**To Reproduce**
1. Go to home
2. Click on any of the menu on the left
3. You will see a `menu` pop up

**Expected behavior**
The tooltip should only pop up when hover.

**OpenSearch Version**
3.0.0

**Dashboards Version**
3.0.0
commit id: a9801a2c31858b84f77e51eab2381ee9744989d9

**Plugins**

None

**Screenshots**

https://github.com/user-attachments/assets/5dc72300-a304-4b46-81e9-21403d7521d2

**Host/Environment (please complete the following information):**
 - Browser and version [e.g. 22]

**Additional context**

### PR DESCRIPTION
**Pull request by danieldong51**
### Description

Reverts addition of EuiTooltip for header menu button, due to bug where tooltip will always pop up when menu is closed or user navigates to another page.

### Issues Resolved

#7483

## Screenshot

| Before (v7Light) | After (v7Light) | Before (v8dark) | After (v8dark) |
| ----- | ----- | ----- | ----- |
| Menu v7 Light Before | Menu v7 Light Post | Menu v8 Dark Before | Menu v8 Dark Post |

## Changelog
- skip

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: src/core/public/chrome/ui/header/header.test.tsx [hunk 1]
```diff
@@ -205,4 +206,17 @@ describe('Header', () => {
 
     expect(component.find('.header__toggleNavButtonSection').exists()).toBeFalsy();
   });
+
+  it('toggles primary navigation menu when clicked', () => {
```
Comment (virajsanghvi) [inline src/core/public/chrome/ui/header/header.test.tsx:210]: was this just for coverage?

**Review by virajsanghvi — DISMISSED** [review]
Approving, assuming checks pass/unrelated

**Comment by ashwin-pc** [comment]
@danieldong51 since this change reverts a change that happened in the same release, we dont need a new changeset file, rather we should either update or remove the old changeset here.

File: src/core/public/chrome/ui/header/header.test.tsx (hunk 1 above)
Comment (danieldong51) [reply src/core/public/chrome/ui/header/header.test.tsx:210]: Yes

**Review by virajsanghvi — APPROVED** [review]

**Review by zhongnansu — APPROVED** [review]

**Pull request merged**